packages/server/src/dto/DocumentStoreDTO.ts

```diff
@@ -0,0 +1,77 @@
+import { DocumentStoreStatus } from '../Interface'
+import { DocumentStore } from '../database/entities/DocumentStore'
+import { convertToValidFilename } from '../utils'
+
+export class DocumentStoreDTO {
+    id: string
+    name: string
+    description: string
+    subFolder: string
+    files: any[]
+    whereUsed: any[]
+    createdDate: Date
+    updatedDate: Date
+    status: DocumentStoreStatus
+    chunkOverlap: number
+    splitter: string
+    codeLanguage: string
+    totalFiles: number
+    totalChunks: number
+    totalChars: number
+    chunkSize: number
+
+    constructor() {}
+
+    static fromEntity(entity: DocumentStore): DocumentStoreDTO {
+        let documentStoreDTO = new DocumentStoreDTO()
+
+        Object.assign(documentStoreDTO, entity)
+        documentStoreDTO.id = entity.id
+        documentStoreDTO.name = entity.name
+        documentStoreDTO.description = entity.description
+        documentStoreDTO.subFolder = entity.subFolder
+        if (entity.whereUsed) {
+            documentStoreDTO.whereUsed = JSON.parse(entity.whereUsed)
+        } else {
+            documentStoreDTO.whereUsed = []
+        }
+        if (entity.metrics) {
+            let metrics = JSON.parse(entity.metrics)
+            documentStoreDTO.totalFiles = metrics.totalFiles
+            documentStoreDTO.totalChars = metrics.totalChars
+            documentStoreDTO.totalChunks = metrics.totalChunks
+        }
+        documentStoreDTO.status = entity.status
+        if (entity.files) {
+            documentStoreDTO.files = JSON.parse(entity.files)
+        }
+        return documentStoreDTO
+    }
+
+    static fromEntities(entities: DocumentStore[]): DocumentStoreDTO[] {
+        return entities.map((entity) => this.fromEntity(entity))
+    }
+
+    static toEntity(body: any): DocumentStore {
```

Comment:
whenever we `updateDocumentStore`, this method is called, and everything is then reset like `files`, `whereUsed` etc

Reply:
I didnt see `whereUsed` being used anywhere in other places, whats the purpose of that?

Reply:
> whenever we `updateDocumentStore`, this method is called, and everything is then reset like `files`, `whereUsed` etc

Fixed.

Reply:
whereUsed is part of functionality in progress. Will hold an array of chatflow id that use this DocumentStore. Will be used to warn the user when they edit / add files / change chunking config on an existing doc store